clippy_lints/src/mem_replace.rs

```diff
@@ -135,33 +135,66 @@ fn check_replace_option_with_none(cx: &LateContext<'_, '_>, src: &Expr<'_>, dest
     }
 }
 
-fn check_replace_with_uninit(cx: &LateContext<'_, '_>, src: &Expr<'_>, expr_span: Span) {
-    if let ExprKind::Call(ref repl_func, ref repl_args) = src.kind {
+fn check_replace_with_uninit(cx: &LateContext<'_, '_>, src: &Expr<'_>, dest: &Expr<'_>, expr_span: Span) {
+    if let ExprKind::MethodCall(method_path, _, method_args) = src.kind {
```

Comment:
This `if let` can be moved down into the `if_chain` to remove one level of indentation

Reply:
Okay, I have moved the `MethodCall` check inside the `if_chain`